BraintreeCore/src/main/java/com/braintreepayments/api/DeviceInspector.kt

```diff
@@ -0,0 +1,156 @@
+    private val appHelper: AppHelper,
+    private val uuidHelper: UUIDHelper,
+    private val signatureVerifier: SignatureVerifier,
+    private val runtime: Runtime,
+    private val superUserApkFile: File
+) {
+    constructor() : this(
+        AppHelper(),
+        UUIDHelper(),
+        SignatureVerifier(),
+        Runtime.getRuntime(),
+        File("/system/app/Superuser.apk")
+    )
+
+    @VisibleForTesting
+    fun getDeviceMetadata(
+        context: Context?,
+        sessionId: String?,
+        integration: String?,
+    ): DeviceMetadata {
+        return DeviceMetadata(
+            platform = "Android",
+            platformVersion = Build.VERSION.SDK_INT.toString(),
+            sdkVersion = BuildConfig.VERSION_NAME,
+            merchantAppId = context?.packageName,
+            merchantAppName = getAppName(context),
+            deviceManufacturer = Build.MANUFACTURER,
+            deviceModel = Build.MODEL,
+            devicePersistentUUID = uuidHelper.getPersistentUUID(context),
+            isSimulator = isDeviceEmulator,
+            sessionId = sessionId,
+            integration = integration,
+            networkType = getNetworkType(context),
+            userOrientation = getUserOrientation(context),
+            appVersion = getAppVersion(context),
+            dropInVersion = dropInVersion,
+            isPayPalInstalled = isPayPalInstalled(context),
+            isVenmoInstalled = isVenmoInstalled(context)
+        )
+    }
+
+    /**
+     * @param context A context to access the installed packages.
+     * @return boolean depending on if the Venmo app is installed, and has a valid signature.
+     */
+    fun isVenmoAppSwitchAvailable(context: Context?): Boolean {
+        val isVenmoIntentAvailable = appHelper.isIntentAvailable(context, venmoIntent)
+        val isVenmoSignatureValid = signatureVerifier.isSignatureValid(
+            context, VENMO_APP_PACKAGE, VENMO_BASE_64_ENCODED_SIGNATURE
+        )
+        return isVenmoIntentAvailable && isVenmoSignatureValid
+    }
+
+    fun isPayPalInstalled(context: Context?): Boolean {
+        return appHelper.isAppInstalled(context, PAYPAL_APP_PACKAGE)
+    }
+
+    fun isVenmoInstalled(context: Context?): Boolean {
+        return appHelper.isAppInstalled(context, VENMO_APP_PACKAGE)
+    }
+
+    private val isDeviceEmulator: Boolean
+        get() = "google_sdk".equals(Build.PRODUCT, ignoreCase = true) ||
+                "sdk".equals(Build.PRODUCT, ignoreCase = true) ||
+                "Genymotion".equals(Build.MANUFACTURER, ignoreCase = true) ||
+                Build.FINGERPRINT.contains("generic")
+
+    private fun getAppName(context: Context?): String =
+        context?.let {
+            getApplicationInfo(it)?.let { appInfo ->
+                context.packageManager.getApplicationLabel(appInfo).toString()
+            }
+        } ?: "ApplicationNameUnknown"
+
+    private fun getApplicationInfo(context: Context) =
```

Comment:
```suggestion
    private fun getApplicationInfo(context: Context?) =
```
🧶 : We could clean up other `context?.let {}` invocations with this style. Take or leave. We'd end up with some duplication but it may make the code more readable.